### PR TITLE
REF: FloatButtons component

### PR DESCRIPTION
**Pull request by limpbrains**
Problem: 
There are 3 very alike components: 
- Scan button on main screen
- Send/recieve buttons on wallet/list screen
- Undo/save on provide entropy screen
All of them looks the same
But not all of them working correctly with different languages.

Solution:
Create universal component with nice UI that supports any number of buttons. Also, they should work for any language.

Before:
image
image

After:
image
image

It also works with 3 buttons:
image

Two buttons with different content width will still have the same size
image

To make it happen it firsts render buttons outside the screen to calculate their sizes. Then calculate container width render them on screen

API:
```js

import { FContainer, FButton } from '../../components/FloatButtons';
<FContainer>
  <FButton
    onPress={() => console.log('down')}
    icon={<Icon name="arrow-down"/>}
    text={loc.receive.down}
  />
  <FButton
    onPress={() => console.log('up')}
    icon={<Icon name="arrow-up"/>}
    text={loc.receive.up}
  />
</FContainer>
```

### CODE REVIEW

**Comment by limpbrains** [comment]
Little problem:
Before there was one place where `size` for `<Icon` was calculated. In `BlueComponents.js`
Now because Icon provided as a paratemeter to `FButton` there are 3:
FloatButtons, screen/provideEntropy, screen/wallet/transactions
Can we move such calculations to `BlueDefaultTheme` ? @marcosrdz what do you think?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/de2dvfwrchuwjnxdacpbehpenr

[download apk](http://filestorage.bluewallet.io:1488/1599507821.apk)

**Comment by marcosrdz** [comment]
Conflicts

**Comment by limpbrains** [comment]
@marcosrdz fixed

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/d3fqkrxdnxxru67f2vzfp1m4er

[download apk](http://filestorage.bluewallet.io:1488/1599677386.apk)

**Comment by marcosrdz** [comment]
@limpbrains  button doesnt highlight when tapping, only text does.

**Comment by Overtorment** [comment]
indeed button doesnt highlight tap. also, text is visibly cropped on appetize:

image

File: screen/wallets/transactions.js [hunk 1]
```diff
@@ -630,6 +666,12 @@ const WalletTransactions = () => {
       });
     }
   };
+  console.info('RECIEVE', wallet.allowReceive());
+
```
Comment (Overtorment) [inline screen/wallets/transactions.js:670]: debug prints

**Review by Overtorment — CHANGES_REQUESTED** [review]
mostly lgtm

File: components/FloatButtons.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+    if (last) {
+      style.paddingRight = BORDER_RADIUS;
+      totalWidth += BORDER_RADIUS;
+    }
+    style.width = totalWidth;
+  }
+
+  return (
+    <TouchableOpacity style={[bStyles.root, style]} {...props}>
```
[the remark this answers is not in the document]
Comment (limpbrains) [reply components/FloatButtons.js:102]: Let's try when we will have RN 0.63, we are on 0.62 currently

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rdq1n2m68m7yj6cuhg996cx1kw

[download apk](http://filestorage.bluewallet.io:1488/1600600073.apk)

**Comment by limpbrains** [comment]
@marcosrdz @Overtorment all is fixed, please review again

**Review by Overtorment — APPROVED** [review]
lgtm

**Review by ncoelho — CHANGES_REQUESTED** [review]
- There a crash when opening watch-only
- Scan size seems very small in english
- The buttons are loading after the screen loads, is this on purpose?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/7ubuxu8hjxyu1xuugepc27xhg4

[download apk](http://filestorage.bluewallet.io:1488/1600683143.apk)

**Comment by limpbrains** [comment]
>     * There a crash when opening watch-only
fixed
>     * Scan button size seems very small in english, are the paddings the same as before?
fixed for Scan button
>     * The buttons are loading after the screen loads, is this on purpose?
Not on purpose. This component renders in 2 steps. On the wallet screen there is list of transaction component it blocks JS for a second, this prevents FloatButtons to appear.
Do you think this is critical ?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9005ykk3kuv1rb0txc5k399z0w

[download apk](http://filestorage.bluewallet.io:1488/1600684348.apk)

**Comment by ncoelho** [comment]
> Do you think this is critical ?

Not at all, just asking because it is noticeable on slower devices. Making it look slower is never good.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/13nkp3b8t7r07nxbnqq5wc4vjm

[download apk](http://filestorage.bluewallet.io:1488/1600947255.apk)

**Review by marcosrdz — APPROVED** [review]

**Comment by ncoelho** [comment]
Check padding on two buttons case. Should be:

_button1_ + _button2_

Screenshot 2020-09-30 at 17 21 34

**Comment by limpbrains** [comment]
Because buttons have different content length I can't use grid. 

Master branch. Here content of Send button is center aligned:
изображение

I've increased button paddings in the middle, so it is mo accurate now. My branch:
изображение

Very long send button example:
изображение

**Comment by ncoelho** [comment]
Looks good, thanks for checking all the details.

**Review by ncoelho — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xdwtfun8jg6vn6f5dnpntbg75m

[download apk](http://filestorage.bluewallet.io:1488/1601633630.apk)

**Review by Overtorment — CHANGES_REQUESTED** [review]
pls fix conflicts

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/52qkbbjhv917cdyueydzjqnxcm

[download apk](http://filestorage.bluewallet.io:1488/1602144486.apk)

**Comment by limpbrains** [comment]
> pls fix conflicts

fixed

**Pull request merged**